### PR TITLE
LPD-42742 Style book unrestricted File Upload Vulnerability

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Sending here as part of the deal between Platform Experience and BPM to receive SF review on backend PRs.

--

LPD: https://liferay.atlassian.net/browse/LPD-42742
LPP: https://liferay.atlassian.net/browse/LPP-56829

# Issue
Style book preview upload does not validate file type, which allows uploading any file to the server.

# Fix
Validate file extension and mime type.

### CODE REVIEW

File: modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/portlet/action/UpdateStyleBookEntryPreviewMVCActionCommand.java [hunk 1]
```diff
@@ -44,65 +50,86 @@ protected void doProcessAction(
...
+
+				serviceContext.setAddGroupPermissions(true);
+				serviceContext.setAddGuestPermissions(true);
+
+				repository = PortletFileRepositoryUtil.addPortletRepository(
+					themeDisplay.getScopeGroupId(),
+					StyleBookPortletKeys.STYLE_BOOK, serviceContext);
+			}
+
+			String fileName = styleBookEntryId + "_preview." + extension;
```
Comment (linolaoj) [inline modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/portlet/action/UpdateStyleBookEntryPreviewMVCActionCommand.java:96]: Hi @thiago-buarqque , can we use the `StringBundler` here?

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java [hunk 2]
```diff
@@ -93,153 +99,126 @@ public void setUp() throws Exception {
...
-			new MockLiferayPortletActionRequest();
-
-		StyleBookEntry styleBookEntry =
-			_styleBookEntryLocalService.addStyleBookEntry(
-				null, TestPropsValues.getUserId(), _group.getGroupId(), false,
-				StringPool.BLANK, RandomTestUtil.randomString(),
-				StringPool.BLANK, RandomTestUtil.randomString(),
-				_serviceContext);
+	private String _getFileName(String extension, long styleBookEntryId) {
+		return styleBookEntryId + "_preview." + extension;
```
Comment (linolaoj) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java:172]: @thiago-buarqque , can we use the `StringBundler` here?

**Review by linolaoj — CHANGES_REQUESTED** [review]

**Review by linolaoj — APPROVED** [review]

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:platform-experience

**Comment by thiago-buarqque** [comment]
ci:reevaluate:1367401_6962

**Comment by thiago-buarqque** [comment]
ci:test:platform-experience

**Comment by thiago-buarqque** [comment]
ci:forward

**Pull request closed without merging**